src/scripts/main.js

```diff
@@ -1 +1,105 @@
 'use strict';
+
+const firstPromise = new Promise((resolve, reject) => {
+  let isResolved = false;
```

Comment:
The variable 'isResolved' is redundant because the promise itself maintains the state and cannot resolve or reject more than once. You can remove this variable and rely on the promise's behavior.